stream_alert/rule_processor/sink.py

```diff
@@ -13,121 +13,88 @@
-            self.publish_message(json_dump(sns_dict))
-
-    def _get_sns_topic_arn(self):
-        """Return a properly formatted SNS ARN.
-
-        Args:
-            region: Which AWS region the SNS topic exists in.
-            topic: The name of the SNS topic.
-        """
-        topic = self.env['lambda_function_name'].replace('_streamalert_rule_processor',
-                                                         '_streamalerts')
-
-        return 'arn:aws:sns:{region}:{account_id}:{topic}'.format(
-            region=self.env['lambda_region'],
-            account_id=self.env['account_id'],
-            topic=topic
-        )
-
-    @staticmethod
-    def _sns_message_size_check(message):
-        """Verify the SNS message is less than or equal to 256KB (SNS Limit)
-        Args:
-            message: A JSON string containing an alert to send to SNS.
-
-        Returns:
-            Boolean result of if the message is within the size constraint
-        """
-        message_size = sys.getsizeof(message)
-        return 0 < message_size <= _SNS_MAX_SIZE
-
-    def publish_message(self, message):
-        """Emit a message to SNS.
-
-        Args:
-            client: The boto3 client object.
-            message: A JSON string containing a serialized alert.
-            topic: The SNS topic ARN to send to.
-        """
-        if not self._sns_message_size_check(message):
-            LOGGER.error('Cannot publish Alerts, message size is too big!')
-            raise SNSMessageSizeError('SNS message size is too big! (Max: 256KB)')
-
-        try:
-            response = self.client_sns.publish(
-                TopicArn=self.topic,
-                Message=message,
-                Subject='StreamAlert Rules Triggered'
-            )
-        except ClientError:
-            LOGGER.exception('An error occurred while publishing alert to sns')
-            return
-
-        if response['ResponseMetadata']['HTTPStatusCode'] != 200:
-            LOGGER.error('Failed to publish message to sns topic: %s', self.topic.split(':')[-1])
-            return
-
-        if self.env['lambda_alias'] != 'development':
-            LOGGER.info('Published alert to %s', self.topic.split(':')[-1])
-            LOGGER.info('SNS MessageID: %s', response['MessageId'])
+            data = _json_dump(alert)
+
+            try:
+                response = self.client_lambda.invoke(
+                    FunctionName=self.function,
+                    InvocationType='Event',
+                    Payload=data
+                )
+
+            except ClientError as err:
+                LOGGER.exception('An error occurred while sending alert to '
+                                 '\'%s\'. Error is: %s. Alert: %s',
+                                 self.function,
+                                 err.response,
+                                 data)
+                return
+
+            if response['ResponseMetadata']['HTTPStatusCode'] != 202:
```

Comment:
Can you be more specific? What's repetitive about it (and are you referring to the try/except above?)? 

If so, I think an HTTP response of non-202 could possibly occur even if there is no ClientError occurred during invocation.  I could also be off on that.. We could also add `FunctionError` to this error logging.